src/AvaloniaEdit.TextMate/TextEditorModel.cs

```diff
@@ -168,7 +168,7 @@ void DocumentOnUpdateFinished(object sender, EventArgs e)
 
             try
             {
-                InvalidateLineRange(_invalidRange.StartLine, _invalidRange.EndLine);
+                InvalidateLineRange(_invalidRange.StartLine < _documentSnapshot.LineCount ? _invalidRange.StartLine : _document.LineCount - 1, _invalidRange.EndLine < _documentSnapshot.LineCount ? _invalidRange.EndLine : _documentSnapshot.LineCount - 1);
```

Comment:
Could you add a method to clamp the line number for better readability?
```suggestion
    int startLine = Math.Clamp(_invalidRange.StartLine, 0, _documentSnapshot.LineCount - 1);
    int endLine = Math.Clamp(_invalidRange.EndLine, 0, _documentSnapshot.LineCount - 1);
    InvalidateLineRange(startLine, endLine);
```

Reply:
Anyway, the changes are ok. 

I should probably protect the NRE later in TextMateSharp library, just in case the consumer passes invalid line numbers as in this case. Thanks for the PR.